### PR TITLE
[Client-core] Merge Feature/generic sdk core 2 branch to main

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.yml [hunk 1]
```diff
@@ -193,6 +193,7 @@ jobs:
             -TargetDirectory $(Build.ArtifactStagingDirectory)
             -Artifacts ('${{ replace(convertToJson(parameters.ReleaseArtifacts), '''', '`''') }}' | ConvertFrom-Json | Where-Object -Not skipPublishPackage )
             -InformationAction Continue
+        condition: ne('${{parameters.ServiceDirectory}}','client-core')
```
Comment (alzimmermsft) [inline eng/pipelines/templates/jobs/ci.yml:196]: Should remove this condition

File: .vscode/cspell.json [hunk 2]
```diff
@@ -151,6 +151,8 @@
     "sdk/eventhubs/azure-messaging-eventhubs/**",
     "sdk/eventhubs/azure-messaging-eventhubs-stress/Dockerfile",
     "sdk/eventgrid/azure-messaging-eventgrid/**",
+    "sdk/client-core/core/**",
```
Comment (srnagar) [inline .vscode/cspell.json:154]: Are we missing other libraries here? core-json, jdk-httpclient?

File: sdk/template/azure-template-stress/dockerfiles/java11.dockerignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+**/.idea
```
Comment (alzimmermsft) [inline sdk/template/azure-template-stress/dockerfiles/java11.dockerignore:1]: Does this need to be checked in?

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 4]
```diff
@@ -903,7 +904,7 @@
               </offlineLink>
             </offlineLinks>
             <failOnError>true</failOnError>
-            <failOnWarnings>true</failOnWarnings>
+            <failOnWarnings>false</failOnWarnings>
```
Comment (alzimmermsft) [inline sdk/parents/azure-client-sdk-parent/pom.xml:907]: This should be reverted

File: sdk/client-core/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Azure Core shared library for Java
```
Comment (srnagar) [inline sdk/client-core/README.md:1]: This README has a lot of references to Azure. We should clean this up to make in unbranded.

File: sdk/client-core/core-json/README.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
+# Azure JSON shared library for Java
```
Comment (srnagar) [inline sdk/client-core/core-json/README.md:1]: Remove references to Azure.

File: sdk/client-core/core-json/pom.xml [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    <license>
+      <name>The MIT License (MIT)</name>
+      <url>http://opensource.org/licenses/MIT</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <distributionManagement>
+    <site>
+      <id>azure-java-build-docs</id>
```
Comment (srnagar) [inline sdk/client-core/core-json/pom.xml:34]: Do we expect to use the Azure build docs for clientcore libraries?

File: sdk/client-core/core/README.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
+# Azure Core shared library for Java
```
Comment (srnagar) [inline sdk/client-core/core/README.md:1]: Remove references to Azure.

**Review by srnagar — APPROVED** [review]
Update the service directory name to `clientcore` as @alzimmermsft suggested before merging.

**Pull request merged**